docs: add 1.20 release notes (#21280)
backstage/backstage: docs/releases/v1.20.0.md

---
id: v1.20.0
title: v1.20.0
description: Backstage Release v1.20.0
---

These are the release notes for the v1.20.0 release of [Backstage](https://backstage.io/).

A huge thanks to the whole team of maintainers and contributors as well as the amazing Backstage Community for the hard work in getting this release developed and done.

## Highlights

### Support for experimental type build has been removed

The `--experimental-type-build` option is no longer supported by any commands in the Backstage CLI. Existing usage should be migrated to using [subpath exports](https://backstage.io/docs/local-dev/cli-build-system#subpath-exports) instead.

### Experimental support for Vite ⚡

> Due to some concerns about performance in larger monorepo setups and hot-reloading times, we’re looking to replace [Webpack](https://webpack.js.org/) with [Vite](https://vitejs.dev/) longer term, and in this release we’re shipping experimental support for Vite behind a `EXPERIMENTAL_VITE` environment variable flag when running `yarn start` in the frontend.

Reviewer: ```suggestion
Due to some concerns about performance in larger monorepo setups and hot-reloading times, we’re looking to replace [Webpack](https://webpack.js.org/) with [Vite](https://vitejs.dev/) longer term, and in this release we’re shipping experimental support for Vite behind an `EXPERIMENTAL_VITE` environment variable flag when running `yarn start` in the frontend.
```

Please reach out by [creating a GitHub issue](https://github.com/backstage/backstage/issues/new/choose) with any problems that you might be running into when using this experimental feature.

### Scaffolder: `react-jsonschema-form@v5` + Design tweaks 💅

> It’s finally here. We’ve shipped support for `react-jsonschema-form@v5` which **should** be a completely non-breaking change for end users. We’ve also tweaked some of the designs for some of the new Scaffolder Pages. You can read more in the [upgrade to `@rjsf/*` v5 docs](https://backstage.io/docs/features/software-templates/migrating-to-rjsf-v5)

Reviewer: ```suggestion
It’s finally here. We’ve shipped support for `react-jsonschema-form@v5` which **should** be a completely non-breaking change for end users. We’ve also tweaked some of the designs for some of the new Scaffolder Pages. You can read more in the [relevant migration docs](https://backstage.io/docs/features/software-templates/migrating-to-rjsf-v5)
```

### Catalog Presentation API

> The Catalog plugin now uses the `EntityPresentationApi` to decide how entities are rendered in the UI. This includes fetching entities in the background in order to show entities using their `metadata.title`. For more information, including how to customize the presentation, see [the PR](https://github.com/backstage/backstage/pull/17474), (documentation is in the works!).

Reviewer: ```suggestion
The Catalog plugin now uses the `EntityPresentationApi` to decide how entities are rendered in the UI. This includes fetching catalog data in the background in order to show entities using their `metadata.title`. For more information, including how to customize the presentation, see [the PR](https://github.com/backstage/backstage/pull/17474), (documentation is in the works!).
```

just gets rid of a small repetition
